odoo/addons/base/models/res_partner.py

```diff
@@ -240,6 +239,10 @@ def default_get(self, default_fields):
         ('check_name', "CHECK( (type='contact' AND name IS NOT NULL) or (type!='contact') )", 'Contacts require a name'),
     ]
 
+    def _get_street_split(self):
```

Comment:
if we have it in tools, why put it here gain?

Reply:
tools is generic for and street argument
_get_street_split(): calls tools.street_split for the partner street
_get_street_split() base_address_extended override uses the stored fields first